base/set.jl

```diff
@@ -425,11 +425,16 @@ end
 
 """
     allunique(itr) -> Bool
+    allunique(f, xs)
```

Comment:
very nitpicky, but using different names for the iterator seems confusing (even though it's maybe helpful for the current definition of this second form).

Reply:
That is a bit ugly, I think I wanted to say `itr = (f(x) for x in xs)` below... but can say it another way. 

Think it's OK to roll these definitions into the same docstring? They could be separate but probably that's just more text to dig through.

Reply:
See if you like [21961ec](https://github.com/JuliaLang/julia/pull/47679/commits/21961ec86033fb13550f0c05c4240b1cd320ede3)

Reply:
Yes looks good. An alternative could be 
```
""""
    allunique([f=identity], itr) -> Bool

Return `true` if all values from `(f(x) for x in itr)` are distinct when compared with [`isequal`](@ref).
"""
```
(but that makes the simpler case (`allunique(itr)`) more difficult to understand...)

Reply:
Maybe it should warn you that how many times `f` is called is regarded as an implementation detail? It may not be called at all (if length 1) and it may be called even after a non-unique elements has been seen (as `_indexed_allunique(collect(C))` makes the whole array & then checks).

Reply:
724f048 adds such a note.